packages/swap/src/routes/v2/quote.ts

```diff
@@ -0,0 +1,334 @@
+        });
+        throw ServiceError.badRequest('invalid request');
+      }
+
+      const start = performance.now();
+
+      logger.info('received a quote request', { query: req.query });
+      const query = queryResult.data;
+
+      const { srcAsset, destAsset, amount, brokerCommissionBps } = queryResult.data;
+      const boostDepositsEnabled = getBoostSafeMode(srcAsset).catch(() => true);
+
+      if (env.DISABLED_INTERNAL_ASSETS.includes(srcAsset)) {
+        throw ServiceError.unavailable(`Asset ${srcAsset} is disabled`);
+      }
+      if (env.DISABLED_INTERNAL_ASSETS.includes(destAsset)) {
+        throw ServiceError.unavailable(`Asset ${destAsset} is disabled`);
+      }
+
+      const amountResult = await validateSwapAmount(srcAsset, BigInt(query.amount));
+
+      if (!amountResult.success) {
+        throw ServiceError.badRequest(amountResult.reason);
+      }
+
+      const ingressFee = await getIngressFee(srcAsset);
+
+      if (ingressFee === null) {
+        throw ServiceError.internalError(`could not determine ingress fee for ${srcAsset}`);
+      }
+
+      if (ingressFee > amount) {
+        throw ServiceError.badRequest(`amount is lower than estimated ingress fee (${ingressFee})`);
+      }
+
+      let limitOrdersReceived: Awaited<ReturnType<Quoter['getLimitOrders']>> | undefined;
+      try {
+        const usdValue = await getUsdValue(amount, srcAsset).catch(() => undefined);
+
+        const [limitOrders, estimatedBoostFeeBps, pools] = await Promise.all([
+          quoter.getLimitOrders(srcAsset, destAsset, amount),
+          env.DISABLE_BOOST_QUOTING || !(await boostDepositsEnabled)
+            ? undefined
+            : getBoostFeeBpsForAmount({ amount, asset: srcAsset }),
+          getPools(srcAsset, destAsset),
+        ]);
+        limitOrdersReceived = limitOrders;
+
+        const dcaQuoteParams = await getDcaQuoteParams(srcAsset, amount);
+
+        const quoteArgs = {
+          srcAsset,
+          destAsset,
+          swapInputAmount: amount,
+          limitOrders,
+          brokerCommissionBps,
+          pools,
+        };
+
+        const [quote, boostedQuote, dcaQuote, dcaBoostedQuote] = await Promise.all([
+          getPoolQuote(quoteArgs).catch((error) => {
+            if (dcaQuoteParams) {
+              return undefined;
+            }
+            throw error;
+          }),
+          estimatedBoostFeeBps &&
+            getPoolQuote({ ...quoteArgs, boostFeeBps: estimatedBoostFeeBps }).catch(
+              () => undefined,
+            ),
+          dcaQuoteParams &&
+            getPoolQuote({
+              ...quoteArgs,
+              swapInputAmount: dcaQuoteParams.chunkSize,
+            }),
+          dcaQuoteParams &&
+            estimatedBoostFeeBps &&
+            getPoolQuote({
+              ...quoteArgs,
+              boostFeeBps: estimatedBoostFeeBps,
+              swapInputAmount: dcaQuoteParams.chunkSize,
+            }).catch(() => undefined),
```

Comment:
think it would be nicer to pass the quote to `adjustDcaQuote` here insie of `.then(adjustDcaQuote)`

Reply:
would need to adjust `adjustDcaQuote` to accept a single quote, but i think that would make sense in general?